src/formatters/__tests__/jsonFormatter-test.js

```diff
@@ -0,0 +1,90 @@
+import test from "tape"
+import stylelint from "../../index"
+import jsonFormatter from "../jsonFormatter"
+
+const config = {
+  "rules": {
+    "number-leading-zero": "always",
+  },
+}
+
+test("expected warnings against css", t => {
+  stylelint.lint({
+    code: "a { top: .2em; }",
+    config,
+    formatter: jsonFormatter,
+  })
+    .then(function (data) {
+      t.equal(
+        data.output,
+        "[{"
+          + `"source":"${data.results[0].source}",`
+          + "\"deprecations\":[],"
+          + "\"invalidOptionWarnings\":[],"
+          + "\"errored\":true,"
+          + "\"warnings\":["
+          + "{"
+            + "\"line\":1,"
+            + "\"column\":9,"
+            + "\"rule\":\"number-leading-zero\","
+            + "\"severity\":\"error\","
+            + "\"text\":\"Expected a leading zero (number-leading-zero)\""
+          + "}"
+        + "]}]"
```

Comment:
Just add a `fixtures/` directory next to the test, or use the one that's already there if there is one.